%!TEX root = ../main.tex
%******************************
%	 Abbreviations and acronyms
%*****************************

\makeglossary
\newacronym{27K}{27K}{Illumina Infinium HumanMethylation27 array}
\newacronym{450K}{450K}{Illumina Infinium HumanMethylation450 array}
\newacronym{5mC}{5mC}{5-methylcytosine}
\newacronym{aDMPs}{aDMPs}{Differentially methylated positions during ageing}
\newacronym{ASD}{ASD}{Autism spectrum disorder}
\newacronym{ATR-X}{ATR-X}{Alpha thalassemia/mental retardation X-linked syndrome}
\newacronym{aVMPs}{aVMPs}{Variably methylated positions during ageing}
\newacronym{B}{B}{CD19$^+$ B cells}
\newacronym{BMIQ}{BMIQ}{Beta-mixture quantile normalisation}
\newacronym{bp}{bp}{Base pairs}
\newacronym{CCC}{CCC}{Cell composition correction}
\newacronym{CD4T}{CD4T}{CD4$^+$ T cells}
\newacronym{CD8T}{CD8T}{CD8$^+$ T cells}
\newacronym{CG}{CG}{$5'$-cytosine-phosphate-guanine-$3'$}
\newacronym{CGI}{CGI}{CpG island}
\newacronym{CHG}{CHG}{$5'$-cytosine-phosphate-H-phosphate-guanine-$3'$, where H corresponds to adenine, thymine or cytosine}
\newacronym{CHH}{CHH}{$5'$-cytosine-phosphate-H-phosphate-H-$3'$, where H corresponds to adenine, thymine or cytosine}
\newacronym{ChIP-seq}{ChIP-seq}{Chromatin immunoprecipitation and sequencing}
\newacronym{CpG}{CpG}{$5'$-cytosine-phosphate-guanine-$3'$}
\newacronym{CPU}{CPU}{Central processing unit}
\newacronym{CP/QP}{CP/QP}{Constrained projection/quadratic programming}
\newacronym{CRF}{CRF}{Cost Reduction Factor in cuRRBS}
\newacronym{cSEA}{cSEA}{Shannon entropy acceleration for the Horvath's epigenetic clock sites}
\newacronym{CTCF}{CTCF}{CCCTC-binding factor}
\newacronym{cuRRBS}{cuRRBS}{customised Reduced Representation Bisulfite Sequencing}
\newacronym{DHS}{DHS}{DNase Hypersensitive Sites}
\newacronym{DHS-DMCs}{DHS-DMCs}{In cell-type deconvolution strategies, reference probes identified using information from differential methylation and chromatin accessibility}
\newacronym{DMCs}{DMCs}{Differentially methylated cytosines}
\newacronym{DMCTs}{DMCTs}{Differentially methylated cytosines in individual cell types}
\newacronym{DMPs}{DMPs}{Differentially methylated positions}
\newacronym{DMRs}{DMRs}{Differentially methylated regions}
\newacronym{DNA}{DNA}{Deoxyribonucleic acid}
\newacronym{DNAmAge}{DNAmAge}{DNA methylation age i.e. epigenetic age calculated with Horvath's epigenetic clock}
\newacronym{EAA}{EAA}{Epigenetic age acceleration}
\newacronym{EPIC}{EPIC}{Illumina Infinium MethylationEPIC array}
\newacronym{epiTOC}{epiTOC}{epigenetic Timer of Cancer (i.e. the epigenetic mitotic clock)}
\newacronym{EV}{EV}{Enrichment Value in cuRRBS}
\newacronym{EWAS}{EWAS}{Epigenome-wide association studies}
\newacronym{FDR}{FDR}{False discovery rate}
\newacronym{FN}{FN}{False negatives}
\newacronym{FP}{FP}{False positives}
\newacronym{FXS}{FXS}{Fragile X syndrome}
\newacronym{GB}{GB}{Gigabytes}
\newacronym{Gbp}{Gbp}{Giga base pairs}
\newacronym{GC content}{GC content}{Guanine + cytosine content}
\newacronym{GEO}{GEO}{Gene Expression Omnibus repository}
\newacronym{Gran}{Gran}{Granulocytes}
\newacronym{gSEA}{gSEA}{Genome-wide Shannon entropy acceleration}
\newacronym{GWAS}{GWAS}{Genome-wide association studies}
\newacronym{hg19}{hg19}{Reference human genome assembly 19}
\newacronym{hg38}{hg38}{Reference human genome assembly 38}
\newacronym{IDOL}{IDOL}{IDentifying Optimal DNA methylation Libraries, a strategy to build cell-type deconvolution references}
\newacronym{IEAA}{IEAA}{Intrinsic epigenetic age acceleration}
\newacronym{iPSCs}{iPSCs}{Induced pluripotent stem cells}
\newacronym{kb}{kb}{Kilo base pairs}
\newacronym{KNN}{KNN}{$k$-nearest neighbours}
\newacronym{MAE}{MAE}{Mean absolute error (in the context of cell-type deconvolution benchmarking) or median absolute error (in the context of Horvath's epigenetic clock)}
\newacronym{MBD}{MBD}{Methyl-CpG-binding domain}
\newacronym{MEFs}{MEFs}{Mouse embryonic fibroblasts}
\newacronym{Mono}{Mono}{CD14$^+$ monocytes}
\newacronym{NF}{NF}{Theoretical number of fragments sequenced in cuRRBS}
\newacronym{NFC}{NFC}{Normalised fold change}
\newacronym{NK}{NK}{CD56$^+$ natural killer cells}
\newacronym{NRC}{NRC}{Normalised read counts}
\newacronym{NRE}{NRE}{Normalised RNA expression}
\newacronym{NRF1}{NRF1}{Nuclear respiratory factor 1}
\newacronym{OOB}{OOB}{Out-of-band fluorescence intensities in the Infinium I probes of Illumina arrays}
\newacronym{OR}{OR}{Odds ratio}
\newacronym{PBMC}{PBMC}{Peripheral blood mononuclear cells}
\newacronym{PC}{PC}{Principal component}
\newacronym{PCA}{PCA}{Principal component analysis}
\newacronym{PCC}{PCC}{Pearson's correlation coefficient}
\newacronym{pcgtAge}{pcgtAge}{Mitotic age according to the epigenetic mitotic clock (epiTOC)}
\newacronym{PCR}{PCR}{Polymerase chain reaction}
\newacronym{PRC2}{PRC2}{Polycomb repressing complex 2}
\newacronym{QC}{QC}{Quality control}
\newacronym{R}{R}{It can have two meanings: robustness variable in cuRRBS or the R programming language}
\newacronym{R$^2$}{R$^2$}{Coefficient of determination}
\newacronym{RAM}{RAM}{Random-access memory}
\newacronym{Repli-seq}{Repli-seq}{genome-wide analysis of replication timing by sequencing}
\newacronym{RMSE}{RMSE}{Root mean squared error}
\newacronym{RNA}{RNA}{Ribonucleic acid}
\newacronym{RNA-seq}{RNA-seq}{RNA sequencing}
\newacronym{RPC}{RPC}{Robust partial correlations}
\newacronym{RRBS}{RRBS}{Reduced Representation Bisulfite Sequencing}
\newacronym{rRNA}{rRNA}{Ribosomal RNA}
\newacronym{SCC}{SCC}{Spearman's correlation coefficient}
\newacronym{SD}{SD}{Standard deviation}
\newacronym{Sexp}{Sex$_p$}{Sex predicted for a sample using DNA methylation data}
\newacronym{SNP}{SNP}{Single-nucleotide polymorphism}
\newacronym{SQN}{SQN}{Stratified quantile normalisation}
\newacronym{sur}{sur}{Signal of unique reads}
\newacronym{TKO}{TKO}{Triple knockout}
\newacronym{TN}{TN}{True negatives}
\newacronym{TP}{TP}{True positives}
\newacronym{TSS}{TSS}{Transcription start site}
\newacronym{WGBS}{WGBS}{Whole Genome Bisulfite Sequencing}
\newacronym{WTS}{WTS}{Wavelet-transformed signals}

