%!TEX root = ../thesis.tex
%*******************************************************************************
%****************************** Fifth Chapter *********************************
%*******************************************************************************

\chapter{Final remarks} \label{c:5}

\ifpdf
	\graphicspath{{Chapter5/Figs/pdf/}}
\else
	\graphicspath{{Chapter5/Figs/svg/}}
\fi

\epigraph{`Caminante, son tus huellas \\ el camino, y nada más; \\ caminante, no hay camino: \\ se hace camino al andar.}{Antonio Machado, 1912 \cite{Machado}}

\bigskip

The purpose of this work was to advance our understanding of the epigenetic ageing clock in humans. In order to do so, I have analysed data from the epigenetic mark that has been more characterised during human ageing: DNA methylation. It is remarkable how Horvath's clock can work across the entire lifespan (even before birth), in different tissues, developement. Suggestive of the two hypothesis originally framed in introduction....

This work has been focused on expanding our knowledge of the epigenetic ageing clock in humans ...

The very nature of bulk DNA methylation measurements and aDMPs reveals epigenetic heterogeneity in the population of cells. They may be fluctuating and they slowly collapsed to states where more of the other type are observed. 

Discuss more links H3K36/ H3K27 methylation. A particular disturbance of the H3K27 landscape could be the methylation of bivalent regions situated in DMVs.DNA hypomethylation causes a redistribution of polycomb and H3K27me3[Redistribution of H3K27me3 upon DNA hypomethylation results in de-repression of Polycomb target genes] DNMT3A is a tumor suppressor in hematologic malignancies [Dnmt3a loss predisposes murine hematopoietic stem cells to malignant transformation]. The interplay between stem cell differentiation and proliferation and the epigenetic ageing clock is strong in blood. Does this apply to other proliferative tissues? It may be worth to define 2 types of clocks. It is reasonable to think that these tissues would have different strategies to maintain their homeostasis and deal with errors. 
It remains to be seen whether the DNA methylation changes observed occur in all the cell types in the tissue or whether changes in the concentration of specific cell types (e.g. progenitor stem cells) or clones are responsible for them. In this sense, single-cell technologies (specially those that profile the transcriptome and the epigenome simultaneously)  will become instrumental for future mechanistic advances on the epigenetic ageing clock.
Single-cell epigenomics: Recording the past and predicting the future

The idea that the changes in the epigenetic ageing clock are a historical record or 'molecular scars' of the activity of an epigenetic maintenance system that is protective are intriguing [Horvath 2018]. fewer somatic mutations, indicative of more stable genomes, have greater epigenetic age acceleration and vice versa [main horvath and erratum]. Compensatory response observed in some transcriptomic changes in the brain (Melike Donertas, personal communication). 

Genetic oscillations [both reviews]
Oscillatory stuff.
Cytosine modifications exhibit circadian oscillations that are involved in epigenetic diversity and aging
TET and hypomethylation in enhancers. 5mC oxidation by Tet2 modulates enhancer activity and timing of transcriptome reprogramming during differentiation [works for non-proliferative tissue, see review 2]
From review 2:
%in non-proliferating cells may also result from errors in its dynamic maintenance by cycles of TET and DNMT activity, perhaps especially at regions of more open and dynamic chromatin. This view fits with the observation that CpGs whose methylation changes with age partly overlap features where methylation is thought to be particularly dynamic; namely, enhancers and CpGs subject to circadian oscillation (Feldmann et al., 2013, Hon et al., 2014, Lu et al., 2014, Oh et al., 2018, Petkovich et al., 2017, Rulands et al., 2018, Wang et al., 2017). Age-associated decay because of maintenance errors may be exacerbated by age-altered expression of TETs and DNMTs, changes in their targeting, and/or altered levels of essential substrates, such as S-adenosylmethionine (SAM) for DNMTs or α-ketoglutarate (α-KG) for TETs
Towards 50/50 in population for a given site: maximum entropy, random cell states defined.

Property of stem cells in the tissue that expands. Lineage tracing 



At the end of the day: ageing process. 



NSD1 also catalises H4K20 methylation.
https://www.pnas.org/content/106/51/21830
Although unclear.
https://www.cell.com/cell-chemical-biology/fulltext/S1074-5521(13)00424-9
H4K20 methylation seems to be enriched in telomeres
https://academic.oup.com/nar/article/46/5/2347/4816214
Consequences of losing H4K20 methylation in telomeres
Review:
https://academic.oup.com/nar/article/41/5/2797/2414826
NSD2 links dimethylation of histone H3 at lysine 36 to oncogenic programming
Epigenetic stress response to problems in telomeres (not only short length)
Changes in chromatin structure have been shown to occur at dysfunctional telomeres7. These changes are often similar to those exhibited at sites of DNA damage, such as phosphorylation of H2AX, changes in H4K20me2 levels and recruitment of tumour suppressor p53-binding protein 1 (TP53BP1), implying that there is a general 'epigenetic' stress response (see the figure, part c).
%https://www.nature.com/articles/nrg2047